qiskit/circuit/add_control.py

```diff
@@ -250,20 +250,40 @@ def apply_basic_controlled_gate(circuit, gate, controls, target):
             elif theta == 0 and phi == 0:
                 circuit.mcp(lamb, controls, target)
             else:
-                circuit.mcp(lamb, controls, target)
+                circuit.mcrz(lamb, controls, target, use_basis_gates=True)
```

Comment:
If I recall correctly, setting ``use_basis_gates=True`` recursively calls ``transpile``. It's not a problem, but I am wondering whether this can be avoided.

Reply:
done in [f0c7e31](https://github.com/Qiskit/qiskit/pull/13801/commits/f0c7e316da07af87da7e4ff87737f35a5bfa9c36).
I also updated other places.